hooks/pre_commit/42-evr_extension_permissions.rb

```diff
@@ -0,0 +1,31 @@
+# Managed databases will be handled automatically.
+return if local_postgresql?
+
+database = param_value('foreman', 'db_database') || 'foreman'
+config = load_db_config(database)
+
+# If postgres is the owner of the DB, then the permissions will not matter.
+return if config[:username] == 'postgres'
+
+check_evr_owner_sql = "SELECT CASE" \
+                      " WHEN r.rolname = '#{config[:username]}' THEN 0" \
+                      " ELSE 1" \
+                      " END AS evr_owned_by_postgres" \
+                      " FROM pg_extension e" \
+                      " JOIN pg_roles r ON e.extowner = r.oid" \
+                      " WHERE e.extname = 'evr';"
+
+command = pg_sql_statement(check_evr_owner_sql)
+logger.debug "Checking if the evr extension is owned by the postgres user via #{command}"
+output, = execute_command(command, false, true, pg_env(config))
+
+case output&.strip
+when '0'
+  # The evr extension is owned by the foreman DB owner, so we can skip this check.
+  return
+when '1'
+  fail_and_exit("The evr extension is not owned by the #{database} DB owner. Please run the following command to fix it: " \
+                "UPDATE pg_extension SET extowner = (SELECT oid FROM pg_authid WHERE rolname='#{config[:username]}');")
```

Comment:
this updates *all* extensions, not only `evr`!

Reply:
I kept this because it felt like all of the Foreman DB extensions should be owned by Foreman, but that isn't strictly necessary. I can add the WHERE clause.

Reply:
they aren't on a fresh install, so we shouldn't modify them

(the longer version: just doing `UPDATE pg_extension SET extowner = …` is *technically* a super bad idea -- you also gotta update all the data *owned* by that extention, it's just damn luck that we don't have to do it here due to the way how we ended up with the wrong owner to begin with).

Reply:
Makes sense, I've updated it to only update the evr extension.